src/Raven.Client/Json/Serialization/NewtonsoftJson/Internal/BlittableJsonConverterBase.cs

```diff
@@ -240,11 +240,36 @@ private static bool ShouldSimplifyJsonBasedOnType(string typeValue)
             return typeof(IEnumerable).IsAssignableFrom(type.GetGenericTypeDefinition());
         }
 
+        internal class TypeCacheItem : IEquatable<TypeCacheItem>
+        {
+            public string Property;
+            public Type Type;
+
+            public bool Equals(TypeCacheItem other)
+            {
+                if (other == null)
+                    return false;
+                return this.Type == other.Type && this.Property == other.Property;
+            }
+
+            public override int GetHashCode()
+            {
+                return Sparrow.Hashing.HashCombiner.CombineInline(Property.GetHashCode(), Type.GetHashCode());
+            }
+        }
+
+        private static Dictionary<TypeCacheItem, Type> _propertyTypeCache = new();
+
         internal static Type GetPropertyType(string propertyName, Type rootType)
         {
             if (rootType == null)
                 return null;
 
+            Type type = null;
+            var item = new TypeCacheItem() { Property = propertyName, Type = rootType };
+            if (_propertyTypeCache.TryGetValue(item, out var result))
+                return type;
```

Comment:
Why are you returning `null` here (the only option for `type` if you _got_ a value? Shouldn't this be `result`?
```suggestion
                return result;
```

Reply:
Good catch. I actually push that one for Jenkins to run the slow tests. Thats why I havent seen improvements on that one when I run the profiling test.